### PR TITLE
Hours Logging Modal

### PR DESCRIPTION
**Pull request by ZJW-92**
Hours log in button， show project name，project description,  start time and end time.

### CODE REVIEW

File: src/main/java/org/coffeemine/app/spring/view/TopBar.java [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
 
+import com.vaadin.flow.component.button.ButtonVariant;
+import com.vaadin.flow.component.dialog.Dialog;
+import com.vaadin.flow.component.formlayout.FormLayout;
 import com.vaadin.flow.component.html.Div;
 import com.vaadin.flow.component.html.H3;
+import com.vaadin.flow.component.button.Button;
 import com.vaadin.flow.component.orderedlayout.FlexComponent;
 import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+import com.vaadin.flow.component.textfield.TextField;
```
Comment (RuthgerD) [inline src/main/java/org/coffeemine/app/spring/view/TopBar.java]: useless imports

File: src/main/java/org/coffeemine/app/spring/components/HoursLogIn.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    private Button create;
+    private Button cancel;
+
+    public HoursLogIn(){
+        newTask = new Task();
+        dialog = new Dialog();
+        newForm = new FormLayout();
+    }
+
+    public void taskCreating(){
```
Comment (RuthgerD) [inline src/main/java/org/coffeemine/app/spring/components/HoursLogIn.java]: why do i feel like you you straight up copped chens' old work?

File: src/main/java/org/coffeemine/app/spring/components/HoursLogIn.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        TextField taskName = new TextField();
+        DatePicker startTime = new DatePicker();
+        startTime.setValue(LocalDate.now());
+        DatePicker endTime = new DatePicker();
+        endTime.setValue(LocalDate.now());
+        TextArea description = new TextArea();
+        description.setPlaceholder("Please provide project description here");
```
Comment (RuthgerD) [inline src/main/java/org/coffeemine/app/spring/components/HoursLogIn.java]: this dialog isnt even supposed to make a project?

File: src/main/java/org/coffeemine/app/spring/components/HoursLogIn.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+public class HoursLogIn {
+    private Task newTask;
+    private Dialog dialog;
+    private FormLayout newForm;
+    private Button hoursLogIn;
+    private Button create;
+    private Button cancel;
+
+    public HoursLogIn(){
```
Comment (RuthgerD) [inline src/main/java/org/coffeemine/app/spring/components/HoursLogIn.java]: I mean, good effort but this is not how we do things ;-;
lesson of today: do not blindly copy someones' work and always question code quality.

File: src/main/java/org/coffeemine/app/spring/components/HoursLogIn.java (hunk 2 above)
Comment (ZJW-92) [reply src/main/java/org/coffeemine/app/spring/components/HoursLogIn.java]: Chen showed his code to me on Friday so I did the same structure as his. But yes I should rename next time.

File: src/main/java/org/coffeemine/app/spring/components/HoursLogIn.java (hunk 4 above)
Comment (ZJW-92) [reply src/main/java/org/coffeemine/app/spring/components/HoursLogIn.java]: Yes sure, I asked chen his code on Friday and thought his code is clear for me to understand. so I didn't even rename attributes and method.  But I should bear in mind next time.

File: src/main/java/org/coffeemine/app/spring/components/HoursLogIn.java (hunk 3 above)
Comment (ZJW-92) [reply src/main/java/org/coffeemine/app/spring/components/HoursLogIn.java]: Ryan said just to write the project time I thought

File: src/main/java/org/coffeemine/app/spring/components/HoursLogIn.java (hunk 4 above)
Comment (AeroStun) [reply src/main/java/org/coffeemine/app/spring/components/HoursLogIn.java]: His code is clear only after he explains to you.
The quality of Chen's code is unarguably not decent enough

**Pull request merged**